net-misc/ytdl-gui/ytdl_gui-1.0.recipe

```diff
@@ -0,0 +1,79 @@
+SUMMARY="A simple-to-use, cross-platform graphical interface for youtube-dl"
+DESCRIPTION="YouTubeDL-GUI is a cross-platform graphical interface for \
+youtube-dl, a downloader for online media sharing services like YouTube,\
+ Vimeo, SoundCloud, Bandcamp, and more. It is based on Qt and aims to be\
+ simple-to-use, while still providing various options like file formats \
+and quality levels."
+HOMEPAGE="https://github.com/JaGoLi/ytdl-gui"
+COPYRIGHT="2020 JaGoLi"
+LICENSE="GNU GPL v3"
+REVISION="1"
+srcGitTag="1.0"
+SOURCE_URI="https://github.com/JaGoLi/ytdl-gui/archive/$srcGitTag.tar.gz"
+CHECKSUM_SHA256="20d262cd3e5ce77b075e0ffee6b9d0480560856f687e1c63097fc146825166c6"
+SOURCE_FILENAME="ytdl-gui-$portVersion-tar.gz"
+SOURCE_DIR="ytdl-gui-$srcGitTag"
+ADDITIONAL_FILES="ytdl_gui.rdef.in"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	ytdl_gui$secondaryArchSuffix = $portVersion
+	app:YouTubeDL_GUI$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	cmd:youtube_dl$secondaryArchSuffix
+	lib:libcrypto$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+	lib:libQt5Core$secondaryArchSuffix
+	lib:libQt5Gui$secondaryArchSuffix
+	lib:libQt5Network$secondaryArchSuffix
+	lib:libQt5Svg$secondaryArchSuffix
+	lib:libQt5Widgets$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libiconv$secondaryArchSuffix
+	devel:libGL$secondaryArchSuffix
+	devel:libQt5Core$secondaryArchSuffix
+	devel:libQt5Gui$secondaryArchSuffix
+	devel:libQt5Network$secondaryArchSuffix
+	devel:libQt5Svg$secondaryArchSuffix
+	devel:libQt5Widgets$secondaryArchSuffix
+	devel:libssl$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:g++$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:lrelease$secondaryArchSuffix >= 5
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:qmake$secondaryArchSuffix >= 5
+	"
+
+BUILD()
+{
+	make build
+}
+
+INSTALL()
+{
+	mkdir $appsDir
+	mv ./src/ytdl_gui $appsDir/ytdl_gui
```

Comment:
Since the app appers in the Deskbar as "YouTubeDL-GUI" I think the executable should be "YouTubeDL-GUI" as well.
Maybe even just simply "YouTubeDL" ?

Reply:
It would be more preferable to have the executable name be the same as either the package name or the upstream project's shorthand (`ytdl-gui`). I have thought about changing the Deskbar link to "*YouTube Downloader*" to match the window title.  However, that seems to be quite generic as there are other such YouTube downloaders that label themselves as such. If any of the non-web ones get ported to Haiku, then some confusion may ensue.

If others feel that it is ok though, then I would be willing to change the Deskbar link name to match the window title.

Reply:
Tab shows "Youtube Downloader" maybe that's more in line?

Reply:
On whatever name you settle, I suggest to have it be the same for executable and Deskbar link. E.g.  a launcher like QuickLaunch won't find the app when entering "youtube", though the app appears as "YouTubeDL-GUI" in the Application menu of the Deskbar. When running, it appears in the Deskbar as the executable's name, currently "ytdl_gui".

The window title of the app "Youtube Downloader" can differ from the app name IMO. But the above mess should be avoided. :)